app/src/main/java/vn/mbm/phimp/me/Camera2.java

```diff
@@ -289,49 +361,45 @@ public void onClick(View v) {
 		LinearLayout linear = (LinearLayout)view.findViewById(R.id.lnCam);
 		linear.bringToFront();
 		flash = (ImageButton)view.findViewById(R.id.flash);
-
+		flash.setImageResource(R.drawable.flash_on);
 		flash.setOnClickListener(new OnClickListener() {
 			@Override
 			public void onClick(View v) {
 				// TODO Auto-generated method stub
-				//Log.e("Flash",preview.camera.getParameters().getFlashMode());
-				try{
+				// Log.e("Flash",preview.camera.getParameters().getFlashMode());
+				try {
+					adjustIconPositions();
+					// Get camera parameters
 					Camera.Parameters parameters = preview.mCamera.getParameters();
-					if (preview.mCamera.getParameters().getFlashMode().equals(Camera.Parameters.FLASH_MODE_OFF))
-						PhimpMe.flashStatus = 0;
-					else
-					if (preview.mCamera.getParameters().getFlashMode().equals(Camera.Parameters.FLASH_MODE_ON))
-						PhimpMe.flashStatus = 1;
-					else PhimpMe.flashStatus = 2;
-					if (PhimpMe.flashStatus == 0){
-						if (statusScreen == 0)	flash.setImageDrawable(getResources().getDrawable(R.drawable.flash_on_r));
-						else flash.setImageDrawable(getResources().getDrawable(R.drawable.flash_on));
-						parameters.setFlashMode(Camera.Parameters.FLASH_MODE_ON);
-						preview.mCamera.setParameters(parameters);
-						PhimpMe.flashStatus = 1;
-					}else if (PhimpMe.flashStatus == 1)
-					{
-						if (statusScreen == 0)	flash.setImageDrawable(getResources().getDrawable(R.drawable.flash_off_r));
-						else flash.setImageDrawable(getResources().getDrawable(R.drawable.flash_off));
-						parameters.setFlashMode(Camera.Parameters.FLASH_MODE_AUTO);
-						preview.mCamera.setParameters(parameters);
-						PhimpMe.flashStatus = 2;
-					}else
-					{
-						if (statusScreen == 0)flash.setImageDrawable(getResources().getDrawable(R.drawable.flash_auto_r));else
-							flash.setImageDrawable(getResources().getDrawable(R.drawable.flash_auto));
-						parameters.setFlashMode(Camera.Parameters.FLASH_MODE_OFF);
-						preview.mCamera.setParameters(parameters);
-						PhimpMe.flashStatus = 0;
+					// Switch flash icon
+					switch (state) {
+						case FLASH_ON:
+							state = FLASH_OFF;
+							parameters.setFlashMode(Camera.Parameters.FLASH_MODE_OFF);
+							preview.mCamera.setParameters(parameters);
+							break;
+						case FLASH_OFF:
+							state = FLASH_AUTO;
+							parameters.setFlashMode(Camera.Parameters.FLASH_MODE_AUTO);
+							preview.mCamera.setParameters(parameters);
+							break;
+						case FLASH_AUTO:
```

Comment:
Remove this case. I am sorry I didn't highlight it in my other review, the default case does the job.